src/psyclone/tests/test_files/dynamo0p3/invalid_builtins_mod.f90

```diff
@@ -143,8 +143,19 @@ module lfric_builtins_mod
      procedure, nopass :: inc_X_divideby_Y_code
   end type inc_X_divideby_Y
 
-  ! TODO in #1107: Add a check that mixing will only be allowed for the
-  ! built-ins that convert 'real'- to 'integer'-valued fields and vice-versa.
+  !> Invalid built-in that has different data type of its field
```

Comment:
"...that has field arguments of different primitive types and is not a data-type conversion..."

Reply:
That reads better, thanks. Changed.